lib/netext/dialer.go

```diff
@@ -33,34 +35,96 @@ import (
 
 	"github.com/loadimpact/k6/lib"
 	"github.com/loadimpact/k6/lib/metrics"
+	"github.com/loadimpact/k6/lib/types"
 	"github.com/loadimpact/k6/stats"
 )
 
-// dnsResolver is an interface that fetches dns information
-// about a given address.
-type dnsResolver interface {
-	FetchOne(address string) (net.IP, error)
+// DNSResolver is an interface that fetches DNS information about a given host.
+type DNSResolver interface {
+	Fetch(host string) (net.IP, error)
+}
+
+type resolver struct {
+	cache      *dnscache.Resolver
```

Comment:
I don't think we should continue to use this library, at all. It's way of handling TTL is... not good: https://github.com/viki-org/dnscache/blob/c70c1f23c5d84ab39054ec6b418a5ed0fa51cade/dnscache.go#L46-L60

Reply:
`viki-org/dnscache` is gone in the latest changes, let me know how it looks.